x-pack/legacy/plugins/file_upload/public/components/json_upload_and_parse.js

```diff
@@ -65,6 +66,11 @@ export class JsonUploadAndParse extends Component {
     indexPatternResp: '',
   };
 
+  constructor(props) {
+    super(props);
+    initServicesAndConstants(props);
```

Comment:
Why are the initialization of kibana_services tied to rendering a react component? Wouldn't, it be better if this was done in `x-pack/legacy/plugins/file_upload/public/plugin.ts` start method?

Reply:
Looking at it again, I think that would be better. Fixed!